Tidy up components docs. (#1596)
yewstack/yew: docs/concepts/components.md

---
## What are Components?

> Components are the building blocks of Yew. They manage their own state and can render themselves to the DOM. Components are created by implementing the `Component` trait on object (e.g. structs or enums). The `Component`

Reviewer: To stay consistent with "the book" <sup>[[source]](https://doc.rust-lang.org/book/ch10-02-traits.html#implementing-a-trait-on-a-type)</sup>:
```suggestion
Components are the building blocks of Yew. They manage their own state and can render themselves to the DOM. Components are created by implementing the `Component` trait for a type. The `Component`
```

Author: That's definitely clearer.

```

:::tip note
Note that this lifecycle method does not require an implementation and will do nothing by default.
:::

### Update

> Components are dynamic and can register to receive messages which are passed asynchronously. The 
> `update` lifecycle method is called for each message. This allows the component to update itself 

Reviewer: I feel like the original sentence uses too many buzzwords.
```suggestion
Communication with components happens primarily through messages which are handled by the
`update` lifecycle method. This allows the component to update itself 
```

based on what the message was, and determine if it needs to re-render itself. Messages can be sent 

> from HTML elements listeners or be sent by child components, Agents, Services, or Futures.

Reviewer: I feel like these should be joined now
```suggestion
by event listeners, child components, Agents, Services, or Futures.
```

Author: Thanks for the catch (oops on my part)

}
```

The `Message` type is used to send messages to a component after an event has taken place; for 
example you might want to undertake some action when a user clicks a button or scrolls down the 
page. Because components tend to have to respond to more than one event, the `Message` type will

> normally be an enum, where each member of the enumeration is an event to be handled.

Reviewer: ```suggestion
normally be an enum, where each variant is an event to be handled.
```

Author: That's definitely less verbose! 😄
